src/context.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+
+
+const initialArtboardState = {
+  width: 0,
+  height: 0,
+  scale: 1,
+  fontScale: 1,
+};
+
+const ArtboardContext = React.createContext({
+  state: initialArtboardState,
+});
+
+export const useWindowDimensions = () => {
+  const { state } = React.useContext(ArtboardContext);
+  const { width, height } = state || {};
+
+  return { width, height };
+}
+
+const ArtboardPropTypes = {
+  viewport: PropTypes.shape({
+    width: PropTypes.number,
+    height: PropTypes.number,
+    name: PropTypes.string,
```

Comment:
Do you think it'd be useful to add `scale` and `fontScale` here? Even if it doesn't change the sketch output, it could be a nice way to provide a hook for people to change those values and control the context

Reply:
Yes, would be great to pass a custom `scale` and `fontScale` to artboards, to better replicate React Native device-specific behaviour. Do you think this should be added to the viewport prop, rather than create a new prop? Not sure if it could create confusion if people think that the `scale`/`fontScale` is supposed to be handled by Sketch viewports?

Reply:
It feels natural in `viewport` and I guess with some docs it would be ok.

Reply:
I've filtered out `scale` and `fontScale` from `ArtboardRenderer.tsx` so they don't pollute the generated file. Does this look good/readable?

https://github.com/airbnb/react-sketchapp/blob/aec1caa61bb0302c34f432b9ca42f6b531efca5b/src/renderers/ArtboardRenderer.ts#L25-L35

Reply:
why not just
```js
presetDictionary: { 
     allowResizedMatching: 0, 
     offersLandscapeVariant: 1, 

     name: props.viewport.name, 
     width: props.viewport.width, 
     height: props.viewport.height,      
},
```
?